staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go

```diff
@@ -185,6 +185,8 @@ func handleRootHealthz(checks ...HealthzChecker) http.HandlerFunc {
 			return
 		}
 
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
```

Comment:
Can we move this line (and the one below it) to between lines 159 and 160? That should ensure that the content type is set for early returns from our handler (i.e. healthz failures).

Reply:
no, http.Error sets these headers itself

Reply:
Ah, interesting.. 

Would it be worth considering modifying mux directly so that we don't have to set the nosniff each time? Maybe something like 'HandleWithMimeType' which always sets nosniff?

Reply:
Mostly, this seems to be fixing the same issue as #72535, it seems possible we would run into this again in the future unless we set these things automatically.

Reply:
:) y i filed this first and then @tallclair saw it and followed up with #72535 yes, we could have merged both into a single PR

Reply:
What about fixing this at the mux level? Then we would not have to set for each individual path.